src/extension.ts

```diff
@@ -323,16 +323,18 @@ export async function activate(context: vscode.ExtensionContext) {
 
         wakePort = await getPort();
 
-        let wakePath: string = pathToExecutable ?? "wake";
-        if (!pathToExecutable && cwd !== undefined) {
-            wakePath = path.join(cwd, "wake");
+        if (pathToExecutable !== null) {
+            cwd = path.dirname(pathToExecutable);
```

Comment:
Since we're using `shell: true` in `execa` (`spawn` previously) we have a problem with paths containing whitespace. The solution is to always use `cwd` when we're using non-global `wake` executable, i.e. always change the directory to the path containing the `wake` executable. Using the `cwd` option works with whitespace (`cwd` is not passed to the shell in this case).